engine/crates/partial-caching/src/output/cache_merging.rs

```diff
@@ -0,0 +1,157 @@
+//! Handles merging cache responses into the OutputStore
+
+use super::{
+    engine_response::InitialOutput,
+    shapes::{ConcreteShape, Field, ObjectShape, OutputShapes},
+    store::{ObjectId, ValueId, ValueRecord},
+};
+
+impl<'a> InitialOutput<'a> {
+    pub fn merge_cache_entry(&mut self, json: serde_json::Value, shapes: &'a OutputShapes) {
+        let Some(root_object_id) = self.store.root_object() else {
+            // Presumably an error bubbled up to the root, so not much we can do here.
+            return;
+        };
+
+        let root_object_shape = shapes.root();
+        let serde_json::Value::Object(mut object) = json else {
+            todo!("something");
```

Comment:
Intentional?

Reply:
yeah, all the todos in here are intentional - just trying to get the basics working before I figure out error handling and such.